Create Blog Post on Paper Elements being removed (#1216)
home-assistant/developers.home-assistant: blog/2022-02-18-paper-elements.md

---
author: Zack Barett
authorURL: https://github.com/zsarnett
title: "Paper Elements are being removed"
---

In 2022.3, we are removing all of our `paper-dropdown-menu` and `paper-input` elements in the Frontend Repository. This means that any custom card that was previously using these elements in your cards or card editors will no longer render correctly and will require updates to your codebase.

## Why did we do this?

> We are converting our code base from the depracated `paper` elements to the new [Material Web Components](https://github.com/material-components/material-components-web-components) from Google. This is to keep our frontend up to date with the latest components and features.

Reviewer: ```suggestion
We are converting our code base from the deprecated `paper` elements to the new [Material Web Components](https://github.com/material-components/material-components-web-components) from Google. This is to keep our frontend up to date with the latest components and features.
```

## What should custom cards do?
 

> You can replace your paper elements with mwc elements but there is no gaurentee that these elements will be loaded when your users load your editor or cards. As well as no guarentee that we use these elements forever.

Reviewer: We should not suggest this. Instead we should suggest creators to bundle their own components with unique names. This is supported by MWC. See ESP Web Tools as an example: https://github.com/esphome/esp-web-tools/blob/main/src/components/ewt-checkbox.ts

Author: ```suggestion
Custom Card Creators can bundle their own components with their own unique names. This is now supported by MWC. See ESP Web Tools that use this: https://github.com/esphome/esp-web-tools/blob/main/src/components/ewt-checkbox.ts
```

---
author: Zack Barett
authorURL: https://github.com/zsarnett
title: "Paper Elements are being removed"
---

In 2022.3, we are removing all of our `paper-dropdown-menu` and `paper-input` elements in the Frontend Repository. This means that any custom card that was previously using these elements in your cards or card editors will no longer render correctly and will require updates to your codebase.

As previously stated, Custom Cards using the elements used in the Home Assistant Frontend has never been supported

## Why did we do this?

We are converting our codebase from the deprecated `paper` elements to the new [Material Web Components](https://github.com/material-components/material-components-web-components) from Google. This is to keep our frontend up to date with the latest components and features.

## What should custom cards do?
 

> Custom Card Creators can bundle their components with their unique names. This is now supported by MWC. See ESP Web Tools that use this: https://github.com/esphome/esp-web-tools/blob/main/src/components/ewt-checkbox.ts

Reviewer: ```suggestion
Custom Card Creators should bundle the components they rely on. This way they can control the version that is being used. The MWC components support registering their elements with custom names: [example from ESP Web Tools](https://github.com/esphome/esp-web-tools/blob/main/src/components/ewt-checkbox.ts)
```

> This comes back to the [blog post by Bram](https://developers.home-assistant.io/blog/2020/10/02/lazyMoreInfo):
> 
> > Is there a solution to all these problems? So custom cards can provide the same uniform user experience, without the risk of having breaking changes every release?
> 
> > The best solution we see is a set of elements created by the custom card community. This set would have its own namespace that would not collide with that of the elements that Home Assistant uses. All custom cards could use these elements, without the risk of breaking changes.

Reviewer: ```suggestion
```
